From 5d2094e88e2b9e023b1f7eaa688758b6b41e5e3c Mon Sep 17 00:00:00 2001
From: Myrtle Shah <gatecat@ds0.me>
Date: Sun, 23 Jan 2022 20:56:02 +0000
Subject: [PATCH 5/8] riscv: Fix DTB pointer save and restore

For XIP kernels, a1 would be overwritten by the call to __copy_data.

Signed-off-by: Myrtle Shah <gatecat@ds0.me>
---
 arch/riscv/kernel/head.S | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/riscv/kernel/head.S b/arch/riscv/kernel/head.S
index b1ca65abeb1e..b4a9be04d33b 100644
--- a/arch/riscv/kernel/head.S
+++ b/arch/riscv/kernel/head.S
@@ -284,10 +284,12 @@ pmp_done:
 	la sp, _end + THREAD_SIZE
 	XIP_FIXUP_OFFSET sp
 	mv s0, a0
+	mv s1, a1
 	call __copy_data
 
-	/* Restore a0 copy */
+	/* Restore a0, a1 copy */
 	mv a0, s0
+	mv a1, s1
 #endif
 
 #ifndef CONFIG_XIP_KERNEL
-- 
2.34.1

